src/Microsoft.AspNet.OData.Shared/Formatter/Serialization/SelectExpandNode.cs

```diff
@@ -242,30 +253,125 @@ public SelectExpandNode(SelectExpandClause selectExpandClause, IEdmStructuredTyp
         /// </summary>
         public ISet<IEdmFunction> SelectedFunctions { get; private set; }
 
-        private void BuildExpansions(SelectExpandClause selectExpandClause, HashSet<IEdmNavigationProperty> allNavigationProperties)
+        /// <summary>
+        /// Gets the path to property corresponding to the SelectExpandNode. Null for a top-level select expand.
+        /// </summary>
+        public IEdmPathExpression PathToProperty { get; set; }
+
+        /// <summary>
+        /// Gets the property corresponding to the SelectExpandNode. Null for a top-level select expand.
+        /// </summary>
+        public IEdmProperty Property { get; set; }
+
+        private void BuildExpansions(IList<SelectItem> selectedItems, HashSet<IEdmNavigationProperty> allNavigationProperties)
```

Comment:
>IList<SelectItem> [](start = 37, length = 17)

Why is this an IList<SelectItem> rather than an IEnumerable<SelectItem>? Do we ever modify the list?  If not, we should make this an IEnumerable<SelectItem> and not call ToList() on Ln 175.